e2e/nomostest/testpredicates/predicates.go

```diff
@@ -1353,20 +1353,50 @@ func ClusterRoleBindingSubjectNamesEqual(subjects ...string) func(o client.Objec
 }
 
 // ValidateError returns true if the specified errors contain an error
-// with the specified error code and (partial) message.
-func ValidateError(errs []v1beta1.ConfigSyncError, code, message string) error {
+// with the specified error code, (partial) message, and resources.
+func ValidateError(errs []v1beta1.ConfigSyncError, code, message string, resources []v1beta1.ResourceRef) error {
 	if len(errs) == 0 {
 		return fmt.Errorf("no errors present")
 	}
+
+	hasMessage := false
+	hasResources := false
+
 	for _, e := range errs {
 		if e.Code == code {
-			if message == "" || strings.Contains(e.ErrorMessage, message) {
+			hasMessage = message == "" || strings.Contains(e.ErrorMessage, message)
+			hasResources = len(resources) == 0 || errorResourcesEqual(resources, e.Resources)
+
+			if hasMessage && hasResources {
 				return nil
 			}
 		}
 	}
-	if message != "" {
-		return fmt.Errorf("error %s not present with message %q: %s", code, message, log.AsJSON(errs))
+
+	if message != "" || len(resources) != 0 {
+		return fmt.Errorf("error %s not present with message %q and resources %v: %s", code, message, resources, log.AsJSON(errs))
 	}
+
 	return fmt.Errorf("error %s not present: %s", code, log.AsJSON(errs))
 }
+
+// errorResourcesEqual checks that the two lists of error resources are equal
+func errorResourcesEqual(want []v1beta1.ResourceRef, got []v1beta1.ResourceRef) bool {
+	if len(want) != len(got) {
+		return false
+	}
+
+	found := make(map[string]bool)
```

Comment:
Suggest using `v1beta1.ResourceRef` as the key type. The name might not be unique